### ISSUE TITLE
nthcomp_lnu_interp ignores kTe_keV: bit-identical output across 1-400 keV, gradient exactly zero (and gamma is clamped dead outside [1.5, 3.5])

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#1822]
## Summary

`nthcomp_lnu_interp(nu, gamma, kTe_keV, kTbb_keV)` ignores `kTe_keV`. The
returned spectrum is **bit-identical** across a 400x range in electron
temperature, at every corner of the other two axes:

```
kTe in (1, 20, 400) keV  ->  sum(L_nu) identical?
   gamma=1.6  kTbb=0.010     True    3.178224e-18
   gamma=1.6  kTbb=0.100     True    4.153737e-18
   gamma=2.0  kTbb=0.010     True    1.714355e-18
   gamma=2.0  kTbb=0.100     True    3.581921e-18
   gamma=3.0  kTbb=0.010     True    1.751800e-19
   gamma=3.0  kTbb=0.100     True    2.489686e-18
```

`d/d(kTe)` is therefore exactly `0.0` everywhere — measured at kTe = 0.5, 1,
2, 5, 10, 50, 100, 200, 500 with gamma=1.8, kTbb=0.01. Not small: exactly
zero. The parameter is accepted, documented as an interpolation axis of a
trilinear table, and has no effect.

The other two axes work. `kTbb_keV` responds monotonically (1.95e-18 at 0.001
-> 4.82e-18 at 0.5), and `gamma` responds on [1.5, 3.5].

kTe sets the exponential rollover in thermal Comptonization, and the probe
grid (nu = 1e12..1e19 Hz, i.e. up to ~41 keV) straddles the rollover for every
kTe tested, so a 400x change cannot legitimately leave the spectrum bit-identical.

## Why no test caught it

`tests/components/agn/test_nthcomp_gradient.py::test_nthcomp_lnu_interp_gradient_near_zero_kTbb`
sweeps kTe = 1, 2, 5 and asserts:

```python
grad = jax.grad(loss)(kte)
assert jnp.isfinite(grad), f"Gradient not finite at kTe={kte_val}: {grad}"
```

Zero is finite. The test is green on a gradient that is identically zero,
which is the exact failure mode the file's own docstring says it exists to
catch ("can produce NaN gradients... tests the fix for the double-where safety
pattern"). Same shape as #1808.

## Second finding in the same measurement: gamma is clamped outside [1.5, 3.5]

```
   gamma= 1.00  f= 4.29871e-18  grad=  0.00000e+00
   gamma= 1.45  f= 4.29871e-18  grad=  0.00000e+00   <- bit-identical
   gamma= 1.49  f= 4.29871e-18  grad=  0.00000e+00   <- bit-identical
   gamma= 1.50  f= 4.29871e-18  grad= -1.60914e-18   <- live from here
   ...
   gamma= 3.49  f= 2.14351e-18  grad= -6.06767e-19
   gamma= 3.50  f= 2.13745e-18  grad=  0.00000e+00
   gamma= 4.00  f= 2.13745e-18  grad=  0.00000e+00   <- bit-identical
```

That is ordinary table clamping and may well be intended, but two things
follow. A fit initialised outside [1.5, 3.5] cannot move — the gradient is
exactly zero, not merely small. And `test_nthcomp_lnu_interp_gradient_at_grid_edges`
evaluates at gamma = 1.3, which is inside the dead zone, while asserting only
`isfinite`. It is named for the grid edges and never reaches a live one.

Whether the clamp should instead raise, or the prior should be bounded to the
table, is the open question. Silently returning a frozen spectrum with a zero
gradient is the failure mode #1521 and #1684 are both about.

## Third: the integration test differentiates a constant

`test_nthcomp_lnu_interp_loss_gradient_finite` builds

```python
observed_lnu = jnp.ones_like(nu)          # 1.0
uncertainty  = jnp.ones_like(nu) * 0.1
chi2 = jnp.sum(((model_lnu - observed_lnu) / uncertainty) ** 2)
```

with `model_lnu ~ 1e-18`. The model is 18 orders of magnitude below the data,
so chi2 = sum((0 - 1)/0.1)^2 = 1.0000e+04 exactly, at **every** gamma —
verified at gamma = 1.2 and 2.2. The "mock likelihood" is a constant and its
gradient is numerical dust (analytic ~1e-16, finite-difference exactly 0).

## Reproduce

```python
import jax, jax.numpy as jnp
from tengri.components.agn._nthcomp import nthcomp_lnu_interp
nu = jnp.linspace(1e12, 1e19, 50, dtype=jnp.float32)
f = lambda kte: jnp.sum(nthcomp_lnu_interp(
        nu, jnp.array(1.8, jnp.float32), kte, jnp.array(0.01, jnp.float32)))
for k in (1.0, 20.0, 400.0):
    x = jnp.asarray(k, jnp.float32)
    print(k, float(f(x)), float(jax.grad(f)(x)))
```

Found while auditing the 53 gradient tests whose every assertion is a
finiteness check (the census in `tests/_grad_parity.py` puts the looser count
at 202 of 367).

**Comment by suchethac** [comment]
Guards for this are now on main (#1739, `bb988c7f4`), so this issue is ratcheted rather than only described.

**The ratchet.** `tests/components/agn/test_nthcomp_gradient.py::TestNthcompGradientStability::test_kte_changes_the_spectrum` is `xfail(strict=True)`. It turns the suite red the moment `kTe_keV` starts doing something, which is what closing this looks like.

**Its control**, in the same class: `test_kte_is_inert_and_the_neighboring_axis_is_not`. A strict xfail can flip green for the wrong reason — someone rewriting the probe band, changing the objective, or perturbing a neighbouring axis by accident. This one pins the failure precisely: `sum(L_nu)` is bit-identical across kTe ∈ (1, 20, 400) keV **while kTbb over (0.001, 0.05, 0.5) is not**, so the inertness is specific to kTe rather than a dead call. If kTe ever varies the spectrum it says so in its own failure message and names this issue.

**The gamma half of the report is also pinned**, since it is separable from the kTe fix:

* `test_gamma_gradient_is_live_inside_the_table[1.7, 2.5, 3.0]` — inside [1.5, 3.5] the gradient must be non-zero *and* match a central difference. The version this replaced asserted only `isfinite` and swept `gamma = 1.3`, which is below the table, in the dead zone — so it was named for the grid edges and never reached a live one.
* `test_gamma_is_frozen_outside_the_table[1.0, 1.3, 4.0]` — outside the table the value is bit-identical to the clamped edge and the gradient is exactly `0.0`. Pinned as measured behaviour, not endorsed: a fit initialised out there cannot move. Whether the clamp should instead raise is the open question in this issue.

**One measurement worth adding**, found after filing: the FD comparison in these tests needs an explicit step. `nthcomp` reads a **float32** lookup table and stays float32 whatever dtype it is handed, so promoting the inputs does not help. Summing 50 samples of a ~1e-18 spectrum leaves roughly three good digits, and no step recovers more — measured relative agreement is 9e-4 at h/x = 1e-3, degrading to 5e-2 by h/x = 1e-5 as cancellation takes over. The tests name 1e-3 as the optimum rather than taking the helper's 1e-5 default.

Related: `tests/_grad_parity.py` now defaults `atol=0.0`. With the old fixed `1e-8` floor, three of the gradient assertions in this file were vacuous — the gradients are ~1e-18, so substituting zero still passed, and the `!= 0.0` assertion beside them was carrying the whole test.
